client/wildcard/src/components/Tooltip/Tooltip.tsx

```diff
@@ -1,63 +1,62 @@
-    children?: ReactNode
+    /** A single child element that will trigger the Tooltip to open on hover. */
+    children: ReactNode
+    /** The text that will be displayed in the Tooltip. */
+    content: string
+    /** The open state of the tooltip when it is initially rendered. */
+    defaultOpen?: boolean
+    /** The preferred side of the trigger to render against when open. Will be reversed if a collision is detected. */
+    placement?: TooltipPrimitive.TooltipContentProps['side']
 }
 
-const TOOLTIP_MODIFIERS: Popper.Modifiers = {
-    flip: {
-        enabled: false,
-    },
-    preventOverflow: {
-        boundariesElement: 'window',
-    },
-}
+/** Arrow width in pixels */
+const TOOLTIP_ARROW_WIDTH = 14
+/** Arrow height in pixel */
+const TOOLTIP_ARROW_HEIGHT = 6
 
-/**
- * Renders a Tooltip that can be positioned relative to a target element.
- *
- * This component should typically only need to be rendered once in a React tree.
- * If you need to attach a tooltip to an specific element, simply add the `data-tooltip` attribute to that element.
- */
-export const Tooltip: React.FunctionComponent<React.PropsWithChildren<TooltipProps>> = ({ className }) => {
-    const { subject, content, subjectSeq, placement = 'auto', delay } = useTooltipState()
-
-    const tooltipStyle = useMemo(() => getTooltipStyle(placement), [placement])
-
-    if (!subject || !content) {
-        return null
-    }
-
-    return (
-        <BootstrapTooltip
-            // Set key prop to work around a bug where quickly mousing between 2 elements with tooltips
-            // displays the 2nd element's tooltip as still pointing to the first.
-            key={subjectSeq}
-            isOpen={true}
-            target={subject}
-            placement={placement}
-            // in order to add our own placement classes we need to set the popperClassNames
-            // here is where bootstrap injects it's placement classes such as 'bs-tooltip-auto' automatically.
-            popperClassName={classNames(styles.tooltip, styles.show, className, tooltipStyle)}
-            arrowClassName={styles.arrow}
-            innerClassName={styles.tooltipInner}
-            // This is a workaround to an issue with tooltips in reactstrap that causes the entire page to freeze.
-            // Remove when https://github.com/reactstrap/reactstrap/issues/1482 is fixed.
-            modifiers={TOOLTIP_MODIFIERS}
-            delay={delay}
-        >
-            {content}
-        </BootstrapTooltip>
-    )
+// Handling the onPointerDownOutside event and preventing the default behavior allows us to keep the Tooltip content open
+// even if the trigger <span> was clicked; this allows buttons to be clicked and text to be selected without dismissing content.
+// Reference: https://github.com/radix-ui/primitives/issues/1077
+function onPointerDownOutside(event: Event): void {
+    event.preventDefault()
 }
+
+export const Tooltip: React.FunctionComponent<TooltipProps> = ({
```

Comment:
It'd be great to add a JSDoc here for:
- Extra docs when developers are using it (e.g. when hovering `Tooltip` in their code)
- To reference future work to consolidate this with the Popover component. I created an issue for this here: https://github.com/sourcegraph/sourcegraph/issues/36080

```suggestion
/**
 * Renders a Tooltip that can be positioned relative to a target element.
 *
 * Our tooltips should:
 *   - Be supplemental to the user journey, not essential.
 *   - Include clear and concise text, that can be understood quickly.
 *   - Avoid any interactive content.
 *
 * Related accessibility guidance: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tooltip_role
 *
 * TODO: We plan to consolidate this logic with our `<Popover />` component in the future.
 * GitHub issue: https://github.com/sourcegraph/sourcegraph/issues/36080
 */
export const Tooltip: React.FunctionComponent<TooltipProps> = ({
```

Reply:
Perfect, thank you!